lib/pbench/agent/tool_data_sink.py

```diff
@@ -186,6 +190,408 @@ def terminate(self):
+            datefmt="%H:%M:%S",
+            level=logging.DEBUG,
+        )
+
+        self.logger = logging.getLogger("pmlogger")
+
+    def read_json(self, json_fn):
+        """ read json file """
+        try:
+            self.logger.info("Reading the pcp mapping file")
+            self.logger.debug("Reading file:%s", json_fn)
+            with open(os.path.join(json_fn)) as json_file:
+                self.mapping = json.loads(json_file.read())
+        except Exception as e:
+            self.logger.critical(
+                "Not able to load the mapping file\n:%s", e, exc_info=True
+            )
+
+    def start_logging(self):
+        """ invoke pmlogger and start logging using pmlogger_check -c """
+        # code to start logging
+        self.logger.info("starting logger")
+
+        command = PCP.pmGetConfig("PCP_BINADM_DIR") + "/pmlogger_check -c {}".format(
+            self.control_file
+        )
+        self.logger.debug("start logger command:%s", command)
+        out = subprocess.Popen(
+            [command], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = out.communicate()
+        self.logger.debug("start logger command stdout:%s", stdout.decode())
+
+        if stdout.decode() != "":
+            # pmlogger failed to start. Log it
+            self.logger.error("not able to start logger successfully")
+        else:
+            self.logger.info("logger started successfully")
+
+    def stop_logging(self):
+        """ stop logging using pmlogger_check -s """
+        # code to stop logging
+        self.logger.info("stop logger")
+
+        command = PCP.pmGetConfig("PCP_BINADM_DIR") + "/pmlogger_check -s -c {}".format(
+            self.control_file
+        )
+        self.logger.debug("stop logger command:%s", command)
+        out = subprocess.Popen(
+            [command], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = out.communicate()
+        self.logger.debug("stop logger command stdout:%s", stdout.decode())
+
+        if stdout.decode() != "":
+            # pmlogger failed to stop. Log it
+            self.logger.error("not able to stop logger successfully")
+        else:
+            self.logger.info("logger stopped successfully")
+
+        # All processes done, do cleanup
+        self.cleanup()
+
+    def check_connection(self, hosts):
+        """ Check if pmcd is running in given list of hosts"""
+        self.logger.info("checking connection")
+
+        for idx, host in enumerate(hosts):
+            res = ""
+            try:
+                self.logger.debug("checking connection on host:%s", host)
+                res = pmapi.pmContext(api.PM_CONTEXT_HOST, host + ":44321")
+            except Exception:
+                res = None
+
+            if res is not None:
+                self.logger.info("Port 44321 is open in:{}".format(host))
```

Comment:
Since we're probably always going to be logging `info` level at least, this isn't a big deal; but it's odd that you're pre-formatting the message string here rather than passing arguments and letting the logger format when/if necessary.